processor/transformer/transformer.go

```diff
@@ -552,6 +552,7 @@ func trackLongRunningTransformation(ctx context.Context, stage string, timeout t
 			log.Errorw("Long running transformation detected",
 				"stage", stage,
 				"duration", time.Since(start).String())
+			return // ensure the loop terminates after logging
```

Comment:
It's bound to get closed already because the context passed is cancelled.
Am I missing something..?

Reply:
If context is not cancelled and transformation is running for more than the configured timeout this loop will get stuck in infinite loop until context is cancelled.

Reply:
I will not be infinite but still the loop with be stuck and there will be a goroutine that will be running and consuming resources until context is cancelled.

Reply:
we could also remove the `for {...}`